File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        start = 1
        minimum = prices[0]
        end = len(prices)
        profit = 0
        while start < end:
            profit_expected = prices[start]-minimum
            profit = max(profit_expected,profit)
            if minimum > prices[start]:
                minimum = prices[start]
            start+=1
        return profit
            

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min_value = prices[0];
        int max_profit = 0;
        for(int i = 1;i < prices.length;i++){
            if(prices[i] < min_value){
                min_value = prices[i];
            }
            else{
                int profit_value = prices[i] - min_value;
                
                max_profit = Math.max(max_profit,profit_value);
            }
        }
        return max_profit;
    }
}